packages/lib-classifier/src/store/SubjectStore.spec.js

```diff
@@ -1,147 +1,150 @@
-
-let rootStore
+import stubPanoptesJs from '../../test/stubPanoptesJs'
 
 const project = ProjectFactory.build()
 const workflow = WorkflowFactory.build({ id: project.configuration.default_workflow })
 const subjects = Factory.buildList('subject', 10)
+const shortListSubjects = Factory.buildList('subject', 2)
 
-const clientStub = {
-  panoptes: {
-    get () {
-      return Promise.resolve({
-        body: {
-          subjects
-        }
-      })
-    }
-  }
-}
+const clientStub = stubPanoptesJs({ subjects, workflows: workflow })
 
 describe('Model > SubjectStore', function () {
-  function setupStores (rootStore) {
-    sinon.stub(rootStore.classifications, 'createClassification')
-    rootStore.projects.setResource(project)
-    rootStore.workflows.setResource(workflow)
-    rootStore.workflows.setActive(workflow.id)
+  function setupStores(panoptesClientStub = clientStub) {
+    const store = RootStore.create({
+      classifications: {},
+      dataVisAnnotating: {},
+      drawing: {},
+      feedback: {},
+      fieldGuide: {},
+      subjectViewer: {},
+      tutorials: {},
+      workflowSteps: {},
+      userProjectPreferences: {}
+    }, {
+        client: panoptesClientStub,
+        authClient: { checkBearerToken: () => Promise.resolve(), checkCurrent: () => Promise.resolve() }
+    })
+    sinon.spy(store.subjects, 'populateQueue')
+
+    store.projects.setResource(project)
+    store.projects.setActive(project.id)
+    store.workflows.setResource(workflow)
+    store.workflows.setActive(workflow.id)
+    return store
   }
   describe('Actions > advance', function () {
+    let rootStore
     before(function () {
-      rootStore = RootStore.create(
-        { projects: ProjectStore.create(), subjects: SubjectStore.create(), workflows: WorkflowStore.create() },
-        { client: clientStub }
-      )
+      rootStore = setupStores()
     })
 
-    it('should make the next subject in the queue active when calling `advance()`', function (done) {
-      setupStores(rootStore)
-      rootStore.subjects.populateQueue().then(() => {
-        expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
-        rootStore.subjects.advance()
-        expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
-        expect(rootStore.subjects.resources.get('1')).to.be.undefined()
-      }).then(done, done)
+    it('should make the next subject in the queue active when calling `advance()`', function () {
+      expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
+      rootStore.subjects.advance()
+      expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
+      expect(rootStore.subjects.resources.get('1')).to.be.undefined()
     })
 
     describe('with less than three subjects in the queue', function () {
-      let populateSpy
+      let rootStore
```

Comment:
I missed this before, but `rootStore` here is shadowing `rootStore` in the outer function, which leads to two references with the same name but different values. Probably a bad idea: https://eslint.org/docs/rules/no-shadow.html

This will be true for all functions that declare `rootStore` inside this block.

Reply:
Fixed by #1045.